Feature/shreyagupta112/created user model integrted passport js and made admin get route

shreyagupta112 opened the pull request:
### Administrative Info
#20 

### Changes
What changes did you make?
Created the model.js, auth.js, scure-routes.js, routes.js, and admin.js files. Model.js is the user schema, admin.js contains a get route which prints out a list of admin or non-admin users, and the other three files deal with authentication.

Testing

### Testing
How did you confirm your changes worked?
I created a few fake users to test out the admin get route and authentication and got it to work locally.

### Confirmation of Change 
Upload a screenshot, if possible. Otherwise, please provide instructions on how to see the change.
I created these files, and haven't submitted any pull requests yet, so the entirety of the files are a new change.

shreyagupta112 committed:
User Schema and Admin get route

User Schema and get route for admin and non-admin users

shreyagupta112 committed:
Created admin.js

Get route which retrieves a list of all admin users or a list of all non-admin users.

shreyagupta112 committed:
Created User model

shreyagupta112 committed:
Debugging

Removing unused exports and fixing incorrect path name

PatrickBrown1 committed:
added temp work

PatrickBrown1 committed:
fixed linting errors

PatrickBrown1 committed:
updated file structure

PatrickBrown1 committed:
Signup, authentication, and secure routes functional

PatrickBrown1 reviewed (changes requested):
I made a lot of changes to the schemas and routes, so now all of the authorization routes are functional. 

The only route left to do is the get users routes.
Can you change the route name to be /getUsers, and take one query param called admin. This query param should be verified as a boolean. If it is true, then it will return all users who are admins. If it is false, it will return all users who are not admins.

mohakvni commented on backend/models/UserModel.js:
```diff
@@ -0,0 +1,87 @@
+const UserSchema = new Schema({
+  verified: {
+    type: Boolean,
+    //  required: true,
```
I think its better to make verified required

mohakvni reviewed (commented):
Looks good to me, apart from the changes Patrick requested, I have added a comment in UserModel for making verified required.

shreyagupta112 committed:
Made requested changes to UserModel.js and UserRoutes

shreyagupta112 committed:
UserRoutes Update

Modified the get route for admin/ non-admin users to check if req.params.admin is a valid input

shreyagupta112 committed:
Update to UserRoutes

shreyagupta112 committed:
Fix UserRoutes

shreyagupta112 committed:
Delete UserModel_20220127125635.js

shreyagupta112 committed:
Delete UserRoutes_20220128175031.js

shreyagupta112 committed:
Delete UserRoutes_20220128174851.js

shreyagupta112 committed:
Delete UserRoutes_20220128174618.js

shreyagupta112 committed:
Delete UserRoutes_20220128173557.js

shreyagupta112 committed:
Delete UserRoutes_20220128172923.js

shreyagupta112 committed:
Delete UserRoutes_20220127143422.js

shreyagupta112 committed:
Delete UserRoutes_20220127143418.js

shreyagupta112 committed:
Delete UserRoutes_20220127143407.js

shreyagupta112 committed:
Delete UserModel_20220127142734.js

shreyagupta112 committed:
Delete UserRoutes_20220127125635.js

shreyagupta112 committed:
Delete UserRoutes_20220127143037.js

PatrickBrown1 committed:
fixed linting errors

PatrickBrown1 commented:
LGTM, will approve when @mohakvni gets a chance to take a look

mohakvni reviewed (approved):
LGTM!
